### DISCUSSION TITLE
NNConv Example for graph classification vs node classification

### DISCUSSION THREAD

**Discussion by alzaia** [pyg-team/pytorch_geometric#5963] [Q&A] [answered]
Hello, in the `QM9` example using `NNConv` layers for the molecule property task, a `GRU` and a `set2set` layer are used at the end of the pipeline. I understand these last layers are there to go from the node level to the graph aggregation (since it is a graph classification task). How would this change if I needed to use it for a node classification task? Can I simply keep the `NNConv` layers and then add an MLP for the node classification head at the end, or do I still need a `GRU` like layer in there? If yes, what other options are there instead of a `GRU` for this example? Thanks!

**Comment by rusty1s** [comment, marked as answer]
Yes, you should just drop the `set2set` layer. For `GRU` I suggest to try out both (with and without) to see which one performs best.

**Reply by alzaia** [reply]
Thanks @rusty1s for the feedback! In this example, the same learnable layers are reused multiple times during the forward pass. What would be the advantage of that, instead of having multiple different layers for each block? The reuse of the learnable layers (`NNConv` and `GRU`) here would reduce the size of the model, but also reduce the expressivity.

**Reply by rusty1s** [reply]
IMO, it is better to not share GNN layers from my own experiments, but this is what the "Neural Message Passing" paper does and so we have followed their architecture.

**Reply by alzaia** [reply]
Right. Thanks @rusty1s !
